Improvements: Add GRDzhadzha as dependency

ShaunFell opened the pull request:
This PR will remove redundant code and incorporate GRDzhadzha as a dependency in order to leverage their code, add their backgrounds, and reduce maintenance burden

ShaunFell committed:
Removed redundant code and added GRDzhadzha as dependency

ShaunFell committed:
changing variable structs to match GRDzhadzha name definitions

ShaunFell committed:
Set GRDzhadzha as dependency and debugged examples/tests

ShaunFell committed:
update CI workflows to include GRDzhadzha checkout

ShaunFell committed:
clang format

ShaunFell committed:
Merge branch 'main' into feature/GRDzhadzhaIncorporation

KAClough commented:
This all looks great. Some minor suggestions:

- I think ADMFixedBGVars is a duplicate still of a file that exists in GRDzhadzha, so should be removed.
- I've merged your PR in GRDzhadzha now so I think you don’t need KerrSchildNew any more
- FluxExtraction maybe should be moved to utils?
- Maybe rename Matter folder -> ProcaFields?
- Default background -> rename as Minkowski? Could perhaps add this to GRDzhaDzha, along with KerrDeSitter, but I am happy to leave them here for now and maybe we can do this later.

ShaunFell committed:
Removed redundant code
Renamed Matter folder to ProcaFields
Moved FluxExtraction code to Diagnostics folder
Renamed DefaultBackground class to Minkowski

ShaunFell committed:
Renamed Matter folder in GNUMakeFiles to ProcaFields

ShaunFell committed:
renamed all instances of KerrSchildNew to KerrSchild
Renamed DefaultBackground file to Minkowski

ShaunFell committed:
clang format

ShaunFell committed:
clang format

ShaunFell commented:
>     * I think ADMFixedBGVars is a duplicate still of a file that exists in GRDzhadzha, so should be removed.
removed 
>     * I've merged your PR in GRDzhadzha now so I think you don’t need KerrSchildNew any more
removed KerrSchildNew 
>     * FluxExtraction maybe should be moved to utils?
I think it'll be better in Diagnostics/, so I put it there in the recent commit 
>     * Maybe rename Matter folder -> ProcaFields?
good idea, done. 
>     * Default background -> rename as Minkowski? Could perhaps add this to GRDzhaDzha, along with KerrDeSitter, but I am happy to leave them here for now and maybe we can do this later.
done. Yea, we can discuss that at a later point.

squashed and merged!

xuanxu commented:
Just referencing the JOSS review to have it linked there
https://github.com/openjournals/joss-reviews/issues/6888
